### PR TITLE
Extend models to accommodate dynamic mentor applications

### LINKED ISSUE #139: Extend the mentor model to accommodate new application
**Is your feature request related to a problem? Please describe.**
The existing mentor model only provides space for two mentor application questions but it needs to be extended with the new application questions.

**Describe the solution you'd like** 
Either we can hard code and add some extra columns to the existing model or we can find a way around to keep the questions somewhere else and store the answers. 
Questions and wireframes: 
https://sef.discourse.group/t/using-miro-boards-to-get-feedback-on-user-flows/240/13?u=gravewalker

### PR DESCRIPTION
**Pull request by Gravewalker666**
## Purpose

The purpose of this PR is to fix #139 

## Goals

* To extend models to accommodate dynamic mentor applications

## Approach

* Introduced two new models, 
  * Question
  * MentorResponses
* Removed attributes `application` and `prerequisites` from `mentor` with the related service methods and controllers 
* Added four new endpoints and update one existing to be compatible with the above new models
  * Add Questions
  * Get Questions
  * Apply as a mentor (Updated)
  * Get mentor responses
  * Edit Mentor Responses

More Info : [SEF Hive: ScholarX Model Update to Improve the Mentor Application](https://sef.discourse.group/t/scholarx-model-update-to-improve-the-mentor-application/259?u=gravewalker)

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Test environment
- Postman
- Java Oracle 8
- Apache Maven 3.6.3
- macOS Big Sur

## Learning
- https://www.baeldung.com/jpa-many-to-many
- https://stackoverflow.com/questions/46740624/org-springframework-orm-jpa-jpasystemexception-could-not-set-field-value-post
- https://stackoverflow.com/questions/3585034/how-to-map-a-composite-key-with-jpa-and-hibernate

### CODE REVIEW

**Review by Piumal1999 — CHANGES_REQUESTED** [review]
Good job @Gravewalker666 
Added some comments.

File: src/main/java/org/sefglobal/scholarx/service/ProgramService.java [hunk 1]
```diff
@@ -311,51 +327,71 @@ public Mentor getLoggedInMentor(long programId, long profileId)
...
-                throw new BadRequestException(msg);
-            }
+    /**
+     * Retrives the {@link MentorResponse} List of a {@link Mentor}
+     *
+     * @param mentorId which is the id of the {@link Mentor}
+     * @return {@link MentorResponse} list
+     * @throws ResourceNotFoundException if a mentor doesn't exist by the given mentorId
+     */
+    public List<MentorResponse> getMentorResponses(long mentorId) throws ResourceNotFoundException {
```
[the remark this answers is not in the document]
Comment (Gravewalker666) [reply src/main/java/org/sefglobal/scholarx/service/ProgramService.java:356]: Yeah, but it has a different signature.
Check the second part of the `get Mentor Response` in the [hive thread](https://sef.discourse.group/t/scholarx-model-update-to-improve-the-mentor-application/259/3?u=gravewalker)

File: src/main/java/org/sefglobal/scholarx/model/Question.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+@Entity
+@Table(name = "question")
+public class Question {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    @Column(nullable = false, columnDefinition = "TEXT")
+    private String Question;
```
[the remark this answers is not in the document]
Comment (Gravewalker666) [reply src/main/java/org/sefglobal/scholarx/model/Question.java:16]: Good Catch, Thanks

File: src/main/java/org/sefglobal/scholarx/model/MentorResponse.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
+package org.sefglobal.scholarx.model;
+
+import javax.persistence.*;
+import java.io.Serializable;
+
+@Entity
+@Table(name = "mentor_response")
+public class MentorResponse implements Serializable {
```
[the remark this answers is not in the document]
Comment (Gravewalker666) [reply src/main/java/org/sefglobal/scholarx/model/MentorResponse.java:8]: Alright, I'll have a look

[the remark this answers is not in the document]
Comment (Gravewalker666) [reply src/main/java/org/sefglobal/scholarx/model/MentorResponse.java:8]: `BaseScholarxModel` got this generated `Id` column right? But we don't need a generated `Id` for this class ergo it's better to keep it this way right?

File: src/main/java/org/sefglobal/scholarx/model/Question.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (Gravewalker666) [reply src/main/java/org/sefglobal/scholarx/model/Question.java]: Resolved with the commit 230b75e

File: src/main/java/org/sefglobal/scholarx/model/Question.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
+package org.sefglobal.scholarx.model;
+
+import org.sefglobal.scholarx.util.QuestionCategory;
+
+import javax.persistence.*;
+import java.util.List;
+
+@Entity
+@Table(name = "question")
+public class Question {
```
[the remark this answers is not in the document]
Comment (Gravewalker666) [reply src/main/java/org/sefglobal/scholarx/model/Question.java]: Resolved with the commit 230b75e

File: src/main/java/org/sefglobal/scholarx/model/MentorResponse.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Piumal1999) [reply src/main/java/org/sefglobal/scholarx/model/MentorResponse.java:8]: Ah right. Then it's okay.

**Review by Piumal1999 — APPROVED** [review]
Thank you @Gravewalker666 
Sorry for the delay

**Pull request merged**